src/account.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+var Transaction = require('../src/transaction').Transaction
+
+function Account () {
+  this.transactions = []
+}
+
+Account.prototype.recordTransaction = function (amount, date) {
+  var transaction = new Transaction(amount, date);
+  this.transactions.push(transaction)
+};
+
+Account.prototype.madeDeposit = function (amount, date) {
+  validateDate(date);
```

Comment:
this is the input point and I wanted to throw an error message and prevent the transaction from moving further